rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -334,6 +335,9 @@ void MCAPStorage::open_impl(const std::string & uri, const std::string & preset_
     case rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY: {
       relative_path_ = uri;
       input_ = std::make_unique<std::ifstream>(relative_path_, std::ios::binary);
+      if (!input_->is_open()) {
+        throw std::runtime_error(std::strerror(errno));
```

Comment:
if `input_` failed to open the file during construction, the errno is set accordingly? i was expecting we need to check the failbit instead, but it would be better that we can just use errno!

How about adding more information for runtime error exception?

```suggestion
        throw std::runtime_error("Failed to open storage " + relative_path_ + std::strerror(errno));
```

Reply:
1. std::ifstream and errno is a bit complicated: cppreference (hence the C++ standard) don't mention errno at all. However, std::ifstream is usually implemented in terms of fopen/open which in turn is specified to set errno. I think, it is reasonable to realy on this behavior. An alternative could be to set the [execption mask ](https://en.cppreference.com/w/cpp/io/basic_ios/exceptions) of `input_ `. IMO this has the disadvantage that we would need to disable it after that probably. Otherwise there might occour other exceptions in `FileStreamReader`, which might not be expected or not dealt with properly.
2. Regarding the exception message: this exception will be catched here: https://github.com/ros2/rosbag2/blob/rolling/rosbag2_storage/src/rosbag2_storage/impl/storage_factory_impl.hpp#L135 Here, the path is logged already, so there is no need to duplicate this infromation in the exception message.

Reply:
thanks for the explanation! lgtm!